File: server/dotnet/sdk-dotnet-example/Business/Domain/Payments/PaymentDetails/ThreeDSecureResults.cs
namespace Business.Domain.Payments.PaymentDetails;

public class ThreeDSecureResults
{
    public string? AcsTransactionId { get; set; }

    public string? AppliedExemption { get; set; }

    public string? AuthenticationStatus { get; set; }

    public string? Cavv { get; set; }

    public string? ChallengeIndicator { get; set; }

    public string? DsTransactionId { get; set; }

    public string? Eci { get; set; }

    public string? ExemptionEngineFlow { get; set; }

    public string? Flow { get; set; }

    public string? Liability { get; set; }

    public string? SchemeEci { get; set; }

    public string? Version { get; set; }

    public string? Xid { get; set; }
}

File: server/ruby/app/business/domain/payments/payment_details/three_d_secure_results.rb
module Business
  module Domain
    module Payments
      module PaymentDetails
        class ThreeDSecureResults
          attr_accessor :acs_transaction_id, :applied_exemption, :authentication_status,
                        :cavv, :challenge_indicator, :ds_transaction_id, :eci,
                        :exemption_engine_flow, :flow, :liability, :scheme_eci,
                        :version, :xid

          def initialize(
            acs_transaction_id: nil, applied_exemption: nil, authentication_status: nil,
            cavv: nil, challenge_indicator: nil, ds_transaction_id: nil, eci: nil,
            exemption_engine_flow: nil, flow: nil, liability: nil, scheme_eci: nil,
            version: nil, xid: nil
          )
            @acs_transaction_id = acs_transaction_id
            @applied_exemption = applied_exemption
            @authentication_status = authentication_status
            @cavv = cavv
            @challenge_indicator = challenge_indicator
            @ds_transaction_id = ds_transaction_id
            @eci = eci
            @exemption_engine_flow = exemption_engine_flow
            @flow = flow
            @liability = liability
            @scheme_eci = scheme_eci
            @version = version
            @xid = xid
          end

          def to_h
            {
              'acsTransactionId' => @acs_transaction_id,
              'appliedExemption' => @applied_exemption,
              'authenticationStatus' => @authentication_status,
              'cavv' => @cavv,
              'challengeIndicator' => @challenge_indicator,
              'dsTransactionId' => @ds_transaction_id,
              'eci' => @eci,
              'exemptionEngineFlow' => @exemption_engine_flow,
              'flow' => @flow,
              'liability' => @liability,
              'schemeEci' => @scheme_eci,
              'version' => @version,
              'xid' => @xid
            }
          end
        end
      end
    end
  end
end